src/test/java/seedu/address/logic/commands/ListPersonsCommandTest.java

```diff
@@ -14,31 +16,53 @@
 public class ListPersonsCommandTest {
 
-    private Model model;
-    private Model expectedModel;
-    private PersonSortField dummySortField;
-
-    @BeforeEach
-    public void setUp() {
-        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
-        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
-        dummySortField = PersonSortField.sortByNoField();
+    @Test
+    public void constructor_nullSortField_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new ListPersonsCommand(null));
     }
 
     @Test
     public void execute_listIsNotFiltered_showsSameList() {
-        String successMessage = String.format(ListPersonsCommand.MESSAGE_SUCCESS, dummySortField.getField());
-        assertCommandSuccess(new ListPersonsCommand(dummySortField), model, successMessage, expectedModel);
+        PersonSortField sortFieldNoField = PersonSortField.sortByNoField();
+        String expectedSuccessMessage = String.format(ListPersonsCommand.MESSAGE_SUCCESS,
+                sortFieldNoField.getField());
+
+        Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+
+        assertCommandSuccess(new ListPersonsCommand(sortFieldNoField), model, expectedSuccessMessage, expectedModel);
     }
 
     @Test
     public void execute_listIsFiltered_showsEverything() {
-        String successMessage = String.format(ListPersonsCommand.MESSAGE_SUCCESS, dummySortField.getField());
+        PersonSortField sortFieldNoField = PersonSortField.sortByNoField();
+        String expectedSuccessMessage = String.format(ListPersonsCommand.MESSAGE_SUCCESS,
+                sortFieldNoField.getField());
+
+        Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+
         showPersonAtIndex(model, INDEX_FIRST_PERSON);
-        assertCommandSuccess(new ListPersonsCommand(dummySortField), model, successMessage, expectedModel);
+        assertCommandSuccess(new ListPersonsCommand(sortFieldNoField), model, expectedSuccessMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_sortByName_showSortedList() {
+        PersonSortField sortFieldName = PersonSortField.createSortField("n");
+
+        Model model = new ModelManager(getUnsortedByNameAddressBook(), new UserPrefs());
+
+        // The typical address book is sorted by name
+        Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+        CommandResult commandResult = new ListPersonsCommand(sortFieldName).execute(model);
+        String expectedSuccessMessage = String.format(ListPersonsCommand.MESSAGE_SUCCESS, sortFieldName.getField());
+
+        assertEquals(expectedSuccessMessage, commandResult.getFeedbackToUser());
+        assertEquals(expectedModel, model);
```

Comment:
`assertCommandSuccess()` does the same thing as these 2 lines. I chose not to use it because it may make the code hard to understand since all 4 args are passed in at the same time. For this test I think splitting the checks into 2 steps makes the code easier to read.

Btw, bypassing `assertCommandSuccess()` and calling `assertEquals()` directly is also done in `AddPersonCommandTest#execute_personAcceptedByModel_addSuccessful()`.